crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -189,17 +190,17 @@ pub async fn app_validation_workflow(
         trigger_integration.trigger(&"app_validation_workflow");
     }
 
+    let validations_dependencies = validation_dependencies.lock();
     Ok(
         // If not all ops have been validated
         // and fetching missing hashes has not timed out,
         // trigger app validation workflow again.
         if outcome_summary.validated < outcome_summary.ops_to_validate
-            && !validation_dependencies
-                .lock()
-                .fetch_missing_hashes_timed_out()
+            && !validations_dependencies.fetch_missing_hashes_timed_out()
         {
-            // Trigger app validation workflow again in 10 seconds.
-            WorkComplete::Incomplete(Some(Duration::from_secs(10)))
+            // Trigger app validation workflow again in 20-2000 milliseconds.
+            let interval = max(101 - validations_dependencies.missing_hashes.len(), 1) * 20;
```

Comment:
There are lots of formulas for this interval. I don't even have enough data to decide if this adaptation to number of missing hashes makes sense or if a fixed interval of 1 second is equally suitable. I'm also up for bringing it down to 10-1000 ms now and bring it up if that causes problems, as this change is also for 0.4.